cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -151,6 +158,20 @@ func CreateNodePoolManager(cloudConfigPath string, discoveryOpts cloudprovider.N
 		nodePoolCache:          newNodePoolCache(&okeClient),
 	}
 
+	// auto discover nodepools from compartments with nodeGroupAutoDiscovery parameter
+	klog.Infof("checking node groups for autodiscovery ... ")
+	for _, arg := range nodeGroupAutoDiscoveryList {
+		nodeGroup, err := nodeGroupFromArg(arg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to construct node group auto discovery from argument: %v", err)
+		}
+		nodeGroup.manager = manager
+		nodeGroup.kubeClient = kubeClient
+
+		manager.nodeGroups = append(manager.nodeGroups, *nodeGroup)
+		autoDiscoverNodeGroups(manager, manager.okeClient, *nodeGroup)
+	}
+
```

Comment:
It seems like node-pools that were _explicitly configured_ via `--nodes` should be added before and take precedent over node-pools that were _discovered_ via `--node-group-auto-discovery`. 

Is that your intent? And what's the expected behavior of, say, the max or min node setting when a pool is specified via `--nodes=2:5:ocid1.nodepool.oc1.np-a` but _also_ discovered via `--node-group-auto-discovery=clusterId:ocid1.cluster.oc1.c-1,compartmentId:ocid1.compartment.oc1..c1,nodepoolTags:cluster-autoscaler-also/enabled=true,min:0,max:10`?

I see you are adding the auto-discovered node-pools _before_ adding the node-pools that were explicitly / statically configured via `--nodes`.

Is that _intentional_?

Reply:
`nodeGroupAutoDiscovery` actually overrides `nodes` parameter with this implementation which means `nodes` parameter is ignored if `nodeGroupAutoDiscovery` is provided.

What I can think of as a solution,

1. We could force the user to provide only one of them in the config, so the CA would fail on startup if both of the parameters were provided and also we would log an error line to state the reason for the end-user to see. The end-user should fix the configuration by removing one of them.
2. If we want both parameters work together, we need to decide which one has a higher priority over the other. I would say `nodes` parameter should override `nodeGroupAutoDiscovery` min/max values.

Please let me know of your thoughts and I will proceed accordingly to make the changes.

Reply:
The _convention_ seems to be to warn against using it in the docs[[1](https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/README.md?plain=1#L64),[2](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md?plain=1#L262)], and/or disallow [[1](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/magnum/magnum_cloud_provider.go#L336)] it in the code.

I'm fine with either documenting it and/or errorring out. As you mentioned, currently the code _quietly_ overrides any static node-pools while also logging messages as it processes each static-node pool, which could cause confusion.

Reply:
I added extra checks to prevent using both parameters together, and also documented it in oci/README.